libraries/chain/database.cpp

```diff
@@ -3898,12 +3897,22 @@ void database::update_virtual_supply()
 
       if( !median_price.is_null() && has_hardfork( STEEM_HARDFORK_0_14__230 ) )
       {
-         auto percent_sbd = uint16_t( ( ( fc::uint128_t( ( dgp.current_sbd_supply * get_feed_history().current_median_history ).amount.value ) * STEEM_100_PERCENT )
+         uint16_t percent_sbd = 0;
+
+         if( has_hardfork( STEEM_HARDFORK_0_21 ) )
+         {
+            percent_sbd = uint16_t( ( ( fc::uint128_t( ( dgp.current_sbd_supply * get_feed_history().current_median_history ).amount.value ) * STEEM_100_PERCENT + dgp.virtual_supply.amount.value/2 )
+               / dgp.virtual_supply.amount.value ).to_uint64() );
+         }
+         else
+         {
+            percent_sbd = uint16_t( ( ( fc::uint128_t( ( dgp.current_sbd_supply * get_feed_history().current_median_history ).amount.value ) * STEEM_100_PERCENT )
```

Comment:
What about the HF20 behavior? It looks like it isn't happening here?

Reply:
The else block is the HF 20 behavior. What about this looks wrong to you?

Reply:
The buffer of 5 for the stop print rate gap that was specific to HF20. And then HF19 had that parameter set to zero. Isn't this going to behave differently for the HF20 blocks?

Reply:
This buffer of 5 was only for the HF21. Not present in HF20 or HF19.

Reply:
Ah that's what I was missing. Thanks!